### PR TITLE
IBX-1689: Fallback for ez ids

### PR DESCRIPTION
**Pull request by GrabowskiM**
| Question                                  | Answer
| ---------------------------------------- | ------------------
| **JIRA issue**                          | [IBX-1689](https://issues.ibexa.co/browse/IBX-1689)
| **Type**                                   | improvement
| **Target Ibexa version** | `v4.0`
| **BC breaks**                          | yes

Changed all eZ/ez- frontend classes/variables to ibexa

Related PRs:

- https://github.com/ibexa/admin-ui/pull/111
- https://github.com/ibexa/calendar/pull/15
- https://github.com/ibexa/commerce-admin-ui/pull/23
- https://github.com/ibexa/commerce-base-design/pull/11
- https://github.com/ibexa/commerce-checkout/pull/10
- https://github.com/ibexa/commerce-fieldtypes/pull/11
- https://github.com/ibexa/commerce-page-builder/pull/11
- https://github.com/ibexa/commerce-price-engine/pull/9
- https://github.com/ibexa/commerce-shop/pull/30
- https://github.com/ibexa/commerce-shop-ui/pull/9
- https://github.com/ibexa/commerce-transaction/pull/17
- https://github.com/ibexa/connector-dam/pull/10
- https://github.com/ibexa/content-forms/pull/11
- https://github.com/ibexa/core/pull/38
- https://github.com/ibexa/fieldtype-matrix/pull/8
- https://github.com/ibexa/fieldtype-page/pull/20
- https://github.com/ibexa/fieldtype-query/pull/7
- https://github.com/ibexa/fieldtype-richtext/pull/8
- https://github.com/ibexa/form-builder/pull/23
- https://github.com/ibexa/image-editor/pull/47
- https://github.com/ibexa/page-builder/pull/44
- https://github.com/ibexa/personalization/pull/23
- https://github.com/ibexa/product-catalog/pull/220
- https://github.com/ibexa/scheduler/pull/28
- https://github.com/ibexa/search/pull/7
- https://github.com/ibexa/segmentation/pull/12
- https://github.com/ibexa/site-factory/pull/13
- https://github.com/ibexa/system-info/pull/11
- https://github.com/ibexa/taxonomy/pull/9
- https://github.com/ibexa/tree-builder/pull/35
- https://github.com/ibexa/user/pull/13
- https://github.com/ibexa/version-comparison/pull/14
- https://github.com/ibexa/workflow/pull/23

#### Checklist:
- [x] Provided PR description.
- [x] Tested the solution manually.
- [x] Checked that target branch is set correctly (main for features, the oldest supported for bugs).
- [x] Asked for a review (ping `@ibexa/engineering`).

### CODE REVIEW

**Review by dew326 — APPROVED** [review]

File: ibexa/commerce/4.0.x-dev/encore/ez.webpack.config.manager.js [hunk 1]
```diff
@@ -1,33 +1,44 @@
 const fs = require('fs');
-const findItems = (eZConfig, entryName) => {
-    const items = eZConfig.entry[entryName];
+const ibexifyEntryName = (entryName) => {
+    const ibexaEntryName = entryName.replace('ezplatform-', 'ibexa-').replace('ezcommerce-', 'ibexacommerce-');
```
Comment (tischsoic) [inline ibexa/commerce/4.0.x-dev/encore/ez.webpack.config.manager.js]: correct me if i'm wrong
```suggestion
    const ibexaEntryName = entryName.replace('ezplatform-', 'ibexa-').replace('ezcommerce-', 'ibexa-commerce-');
```

File: ibexa/commerce/4.0.x-dev/encore/ez.webpack.config.manager.js [hunk 2]
```diff
@@ -1,33 +1,44 @@
 const fs = require('fs');
```
Comment (tischsoic) [inline ibexa/commerce/4.0.x-dev/encore/ez.webpack.config.manager.js:1]: Are we going to rename (ibexify 😉  ) this file?

File: ibexa/commerce/4.0.x-dev/encore/ez.webpack.config.manager.js [hunk 3]
```diff
@@ -1,33 +1,44 @@
...
-    const items = findItems(eZConfig, entryName);
+const replace = ({ ibexaConfig, eZConfig, entryName, itemToReplace, newItem }) => {
+    const config = ibexaConfig ?? eZConfig;
+    const ibexaEntryName = ibexifyEntryName(entryName);
+    const items = findItems(config, ibexaEntryName);
     const indexToReplace = items.indexOf(fs.realpathSync(itemToReplace));
 
     if (indexToReplace < 0) {
-        throw new Error(`Couldn't find item "${itemToReplace}" in entry "${entryName}". Please check if there is a typo in the name.`);
+        throw new Error(`Couldn't find item "${itemToReplace}" in entry "${ibexaEntryName}". Please check if there is a typo in the name.`);
```
Comment (tischsoic) [inline ibexa/commerce/4.0.x-dev/encore/ez.webpack.config.manager.js:37]: Maybe we should note in the message that the name was ibexified?

File: ibexa/commerce/4.0.x-dev/encore/ez.webpack.config.manager.js [hunk 4]
```diff
@@ -1,33 +1,44 @@
 const fs = require('fs');
-const findItems = (eZConfig, entryName) => {
-    const items = eZConfig.entry[entryName];
+const ibexifyEntryName = (entryName) => {
```
Comment (tischsoic) [inline ibexa/commerce/4.0.x-dev/encore/ez.webpack.config.manager.js:6]: Could you explain in a comment, why we need to ibexify entry name? How it is supposed to work?

File: ibexa/commerce/4.0.x-dev/encore/ez.webpack.config.manager.js (hunk 2 above)
Comment (GrabowskiM) [reply ibexa/commerce/4.0.x-dev/encore/ez.webpack.config.manager.js:1]: In the future - yes. ;) Scripts should accept both versions (`ez.xxx.js` and `ibexa.xxx.js`) but it requires backend change and is waiting for Sławek.

**Review by OstafinL — APPROVED** [review]

File: ibexa/commerce/4.0.x-dev/encore/ez.webpack.config.manager.js (hunk 4 above)
Comment (tischsoic) [reply ibexa/commerce/4.0.x-dev/encore/ez.webpack.config.manager.js:6]: Ok, so basically, we changed all our entry names from `ezplatform` to `ibexa`, but clients can have parts of a code in which they extend our entries and still relay on old entries' names. We want to assure that in version 4.0 these old entry names in such extensions still work.

**Review by tischsoic — APPROVED** [review]

**Pull request merged**